db/sqlanalyze.c

```diff
@@ -287,11 +288,29 @@ static int sample_index_int(index_descriptor_t *ix_des)
     return 0;
 }
 
+static inline void set_low_sched_priority() {
+#ifdef _LINUX_SOURCE
+    errno = 0;
+    int rc = nice(20); // set thread low priority -- same as pthread_setschedparam()
+    if (rc < 0 && errno != 0) {
+        logmsg(LOGMSG_WARN, "%s:failed to set nice value:'%s'\n", __func__, strerror(errno));
+        return;
+    }
+    struct sched_param param;
+    int policy;
+    pthread_getschedparam(pthread_self(), &policy, &param);
+    if(param.sched_priority != 0)
+        logmsg(LOGMSG_WARN, "%s:param.sched_priority %d != 0\n", __func__, param.sched_priority);
+#endif
+}
```

Comment:
Did you miss a `pthread_setschedparam` here?

Reply:
Yes, first iteration used pthread_setschedparam, but incidentally all threads have sched_priority of 0, so it seems I would need to change the policy to SCHED_IDLE to achieve something similar to nice().